usbmuxctl/usbmuxctl.py

```diff
@@ -30,9 +30,11 @@ def path_from_usb_dev(dev):
 
     This path is used to find a USB device on the bus or all devices connected to a HUB.
     The path is made up of the number of the USB controller followed be the ports of the HUB tree."""
-    dev_path = ".".join([str(i) for i in dev.port_numbers])
-    dev_path = "{}-{}".format(dev.bus, dev_path)
-    return dev_path
+    if dev.port_numbers:
+        dev_path = ".".join([str(i) for i in dev.port_numbers])
```

Comment:
You can pass an iterator to ''.join(), so wrapping the list comprehension inside the `[…]` is not required.

```python3
".".join(str(i) for i in dev.port_numbers)
```

should work just fine. But that's just nitpicking. Leaving it as-is makes it obvious that adding the `if dev.port_numbers` is the only real change.